Sources/PackageDescription/Version.swift

```diff
@@ -116,3 +116,31 @@ extension Version: CustomStringConvertible {
         return base
     }
 }
+
+// MARK: Enforced Semantic Versioning
+
+enum UpdateType {
+    /// e.g. Remove a method from the public API or change a public method's signature
+    case Major
+    /// e.g. A new method is added to the public API
+    case Minor
+    /// e.g. Change only the implementation of a method
+    case Patch
+}
```

Comment:
The one thing I would consider changing here is the naming of the types to maybe use different names not synonymous with the traditional version component names.

If we do decide we like the Dart style 0.1.0 -> 0.2.0 is a "major" change, then it is probably a good idea to use a different name than "major" for the enum.